### PR TITLE
Change update logic to prevent downgrades

### PR DESCRIPTION
**Pull request by DTMad**

### CODE REVIEW

File: pkg/controller/oneagent/oneagent_utils_test.go [hunk 1]
```diff
@@ -175,7 +177,7 @@ func TestGetPodsToRestart(t *testing.T) {
 	oa := newOneAgent()
 	oa.Status.Version = "1.2.3"
 	oa.Status.Instances = map[string]dynatracev1alpha1.OneAgentInstance{"node-3": {Version: "outdated"}}
-	doomed, err := findOutdatedPodsInstaller(pods, dtc, oa)
+	doomed, err := findOutdatedPodsInstaller(pods, dtc, oa, logf.ZapLoggerTo(os.Stdout, true))
```
Comment (lrgar) [inline pkg/controller/oneagent/oneagent_utils_test.go:180]:
```suggestion
	doomed, err := findOutdatedPodsInstaller(pods, dtc, oa, consoleLogger)
```

File: pkg/controller/oneagent/oneagent_update.go [hunk 2]
```diff
@@ -207,3 +209,43 @@ func (r *ReconcileOneAgent) setVersionByIP(instance dynatracev1alpha1.BaseOneAge
...
+		if err != nil {
+			logger.Error(err, "failed to parse desired version number", "desired", desired)
+			return false
+		}
+
+		if av < dv {
+			return true
+		}
+		if av > dv {
+			logger.Info("downgrade detected! downgrades are not supported")
```
Comment (lrgar) [inline pkg/controller/oneagent/oneagent_update.go:245]: Also an error.

File: pkg/controller/oneagent/oneagent_update.go [hunk 3]
```diff
@@ -207,3 +209,43 @@ func (r *ReconcileOneAgent) setVersionByIP(instance dynatracev1alpha1.BaseOneAge
...
+func isDesiredNewer(actual string, desired string, logger logr.Logger) bool {
+	aa := strings.Split(actual, ".")
+	da := strings.Split(desired, ".")
+
+	for i := 0; i < len(aa); i++ {
+		if i == len(aa)-1 {
+			if aa[i] < da[i] {
+				return true
+			} else if aa[i] > da[i] {
+				logger.Error(nil, "downgrade detected! downgrades are not supported")
```
Comment (lrgar) [inline pkg/controller/oneagent/oneagent_update.go:222]: Does this `nil` not break anything?

Comment (DTMad) [reply pkg/controller/oneagent/oneagent_update.go]: Created a separate error for the downgrade case

**Review by lrgar — APPROVED** [review]

**Pull request merged**